doc/src/manual/performance-tips.md

```diff
@@ -257,9 +257,16 @@ better than `IdDict{Type, Vector}`
 
 See also the discussion under [Parametric Types](@ref).
 
+## Type annotations
+
+In many languages with optional type annotations, adding such annotations is the principal way to
+make code run faster. This is *not* the case in Julia. In Julia, the compiler generally knows
+the types of all function arguments, local variables, and expressions. However, there are a few
+specific instances where annotations are helpful.
+
 ### Avoid fields with abstract type
```

Comment:
This is already present here:
https://github.com/JuliaLang/julia/blob/4a6ada6eb36eaeb9bde9b94c01bcf6f0e3ce7ae6/doc/src/manual/performance-tips.md?plain=1#L209-L214